yaksh/models.py

```diff
@@ -1503,6 +1495,9 @@ class AnswerPaper(models.Model):
 
     objects = AnswerPaperManager()
 
+    class Meta:
+        unique_together = ('user', 'question_paper', 'attempt_number')
```

Comment:
If the same quiz is added to two different courses then this will fail, so I think you need to add course also.